client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -597,6 +598,18 @@ static Request rollover(RolloverRequest rolloverRequest) throws IOException {
         return new Request(HttpPost.METHOD_NAME, endpoint, params.getParams(), entity);
     }
 
+    static Request getSettings(GetSettingsRequest getSettingsRequest) throws IOException {
+        Params params = Params.builder();
+        params.withIndicesOptions(getSettingsRequest.indicesOptions());
+        params.withLocal(getSettingsRequest.local());
```

Comment:
there are a few other params that we need to provide. Unfortunately it's all params that the transport client doesn't support but only the ES REST layer knows about. What we've been doing in these situations is add those params to the request object, and document that only the high-level REST client uses them. We've already encountered flat_settings for instance , and this one also has `include_defaults`.

Reply:
I definitely understand adding include_defaults -- but is there a reason to include flat_settings -- won't the behavior of the java API be identical whether that flag is set or not ?

Reply:
You are right! I thought that somehow we would parse settings differently with or without the flag, but I have just tested it and nothing changes. So, ideally we would support this flag as it is part of our SPEC, but practically it does not do anything given how we parse settings back given that we use the `Settings` class. It would only make sense to expose this flag if we changed the way settings were returned. Thanks for pointing this out. I will also have the remove support for this flag from a couple of other API where we added it :(